Source/Core/Core/HW/WiimoteEmu/MotionPlus.cpp

```diff
@@ -25,27 +28,67 @@ void MotionPlus::Reset()
 
   m_activation_progress = {};
 
+  // Seeing as we allow disconnection of the M+, we'll say we're not integrated.
+  // (0x00 or 0x01)
+  constexpr u8 IS_INTEGRATED = 0x00;
+
   // FYI: This ID changes on activation/deactivation
-  constexpr std::array<u8, 6> initial_id = {0x00, 0x00, 0xA6, 0x20, 0x00, 0x05};
+  constexpr std::array<u8, 6> initial_id = {IS_INTEGRATED, 0x00, 0xA6, 0x20, 0x00, 0x05};
   m_reg_data.ext_identifier = initial_id;
 
-  // Calibration data.
-  // Copied from real hardware as it has yet to be fully reverse engineered.
-  // It's possible a checksum is present as the other extensions have one.
-  constexpr std::array<u8, 32> cal_data = {
-      0x78, 0xd9, 0x78, 0x38, 0x77, 0x9d, 0x2f, 0x0c, 0xcf, 0xf0, 0x31,
-      0xad, 0xc8, 0x0b, 0x5e, 0x39, 0x6f, 0x81, 0x7b, 0x89, 0x78, 0x51,
-      0x33, 0x60, 0xc9, 0xf5, 0x37, 0xc1, 0x2d, 0xe9, 0x15, 0x8d,
-      // 0x79, 0xbc, 0x77, 0xa3, 0x76, 0xd9, 0x30, 0x6c, 0xce, 0x8a, 0x2b,
-      // 0x83, 0xc8, 0x02, 0x0e, 0x70, 0x74, 0xb5, 0x79, 0x8e, 0x76, 0x45,
-      // 0x38, 0x22, 0xc7, 0xd6, 0x32, 0x3b, 0x2d, 0x35, 0xde, 0x37,
+  // Build calibration data.
+
+  // Matching signedness of my real Wiimote+.
+  // This also results in all values following the "right-hand rule".
+  constexpr u16 YAW_SCALE = CALIBRATION_ZERO - CALIBRATION_SCALE_OFFSET;
+  constexpr u16 ROLL_SCALE = CALIBRATION_ZERO + CALIBRATION_SCALE_OFFSET;
+  constexpr u16 PITCH_SCALE = CALIBRATION_ZERO - CALIBRATION_SCALE_OFFSET;
+
+#pragma pack(push, 1)
+  struct CalibrationBlock
+  {
+    u16 yaw_zero = Common::swap16(CALIBRATION_ZERO);
+    u16 roll_zero = Common::swap16(CALIBRATION_ZERO);
+    u16 pitch_zero = Common::swap16(CALIBRATION_ZERO);
+    u16 yaw_scale = Common::swap16(YAW_SCALE);
+    u16 roll_scale = Common::swap16(ROLL_SCALE);
+    u16 pitch_scale = Common::swap16(PITCH_SCALE);
+    u8 degrees_div_6;
+  };
+
+  struct CalibrationData
+  {
+    CalibrationBlock fast;
+    u8 uid_1;
+    Common::BigEndianValue<u16> crc32_msb;
+    CalibrationBlock slow;
+    u8 uid_2;
+    Common::BigEndianValue<u16> crc32_lsb;
   };
-  // constexpr std::array<u8, 32> cal_data = {
-  //     0x7d, 0xe2, 0x80, 0x5f, 0x78, 0x56, 0x31, 0x04, 0xce, 0xce, 0x33,
-  //     0xf9, 0xc8, 0x04, 0x63, 0x22, 0x77, 0x26, 0x7c, 0xb7, 0x79, 0x62,
-  //     0x34, 0x56, 0xc9, 0xa3, 0x3a, 0x35, 0x2d, 0xa8, 0xa9, 0xbc,
-  // };
-  m_reg_data.calibration_data = cal_data;
+#pragma pack(pop)
+
+  static_assert(sizeof(CalibrationData) == 0x20, "Bad size.");
+
+  static_assert(CALIBRATION_FAST_SCALE_DEGREES % 6 == 0, "Value aught to be divisible by 6.");
```

Comment:
Does anybody actually use the word "aught"? :D

Reply:
Damn right we do.

Reply:
I've changed it. :P